### PR TITLE
WOWL: integrate mail systray icons

### PR DESCRIPTION
**Pull request by brboi**
This PR is ready to review.
A fix still needs to get implemented for the actions target new, to render their buttons inside a dialog footer

### CODE REVIEW

File: addons/wowl/static/src/views/legacy/legacy.ts [hunk 1]
```diff
@@ -250,3 +250,31 @@ odoo.define("wowl.legacyViews", async function (require: any) {
...
+
+  class SystrayItemAdapter extends ComponentAdapter {
+    env = Component.env;
+  }
+
+  const legacySystrayMenuItems = legacySystrayMenu.Items as any[];
+  // registers the legacy systray menu items from the legacy systray registry
+  // to the wowl one, but wrapped into Owl components
+  legacySystrayMenuItems.forEach((item, index) => {
+    if (item.prototype.template !== "UserMenu") {
```
Comment (brboi) [inline addons/wowl/static/src/views/legacy/legacy.ts]: A small suggestion to make it super explicit:
```suggestion
  // registers the legacy systray menu items from the legacy systray registry
  // to the wowl one, but wrapped into Owl components
  legacySystrayMenuItems.forEach((item, index) => {
    // blacklisting already wowl converted items
    const blacklist = ["UserMenu"];
    if (!blacklist.includes(item.prototype.template)) {
```

File: addons/wowl/static/src/views/legacy/legacy_view_dialogs.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  patched() {
+    this.updateDialog();
+  }
+
+  updateDialog() {
+    const modal = this.dialogRef.comp.modalRef.el as HTMLElement;
+
+    const dialogFooter = modal.querySelector("footer") as HTMLElement;
+    this.compRef.comp.compRef.comp.widget.renderButtons(dialogFooter);
+
```
Comment (brboi) [inline addons/wowl/static/src/views/legacy/legacy_view_dialogs.ts]: const compFromUpdateUI = this.compRef.comp
    const compFromViewAdapter = compFromUpdateUI.comp

File: addons/wowl/static/src/core/registry.ts [hunk 3]
```diff
@@ -11,8 +11,11 @@
  * 3. it provides a chained API to add items to the registry.
  */
 
+import { Callback } from "@odoo/owl/dist/types/core/event_bus";
```
Comment (aab-odoo) [inline addons/wowl/static/src/core/registry.ts]: not sure it's worth retrieve this from owl... it's just a function after all. And actually, it's more than that, as you know the arguments of this function (key: string, value: any). Btw, what if I do registry.add(2, 'blabla')? Shouldn't this be allowed?

File: addons/wowl/static/src/views/legacy/legacy.ts [hunk 4]
```diff
@@ -250,3 +250,33 @@ odoo.define("wowl.legacyViews", async function (require: any) {
...
+
+  const legacySystrayMenuItems = legacySystrayMenu.Items as any[];
+  // registers the legacy systray menu items from the legacy systray registry
+  // to the wowl one, but wrapped into Owl components
+  legacySystrayMenuItems.forEach((item, index) => {
+    // blacklisting already wowl converted items
+    const blacklist = ["UserMenu"];
+    if (!blacklist.includes(item.prototype.template)) {
+      const name = `_legacy_systray_item_${index}`;
+      const sequence = (index + 1) * 10;
```
Comment (aab-odoo) [inline addons/wowl/static/src/views/legacy/legacy.ts]: the legacy items (can) have a sequence number set on their prototype

File: addons/wowl/static/src/services/action_manager/action_manager.ts [hunk 5]
```diff
@@ -32,6 +32,7 @@ export type ActionDescription = any;
 
 export type ActionRequest = ActionId | ActionXMLId | ActionTag | ActionDescription;
 export interface ActionOptions {
+  additional_context?: { [key: string]: any };
```
Comment (aab-odoo) [inline addons/wowl/static/src/services/action_manager/action_manager.ts]: camelcase variables

**Review by aab-odoo — APPROVED** [review]
Good job overall ! :+1:

File: addons/wowl/static/src/core/registry.ts (hunk 3 above)
Comment (Polymorphe57) [reply addons/wowl/static/src/core/registry.ts]: changed

File: addons/wowl/static/src/services/action_manager/action_manager.ts (hunk 5 above)
Comment (Polymorphe57) [reply addons/wowl/static/src/services/action_manager/action_manager.ts]: done

**Review by aab-odoo — APPROVED** [review]

**Pull request merged**